modular/gater/object_handler.go

```diff
@@ -396,19 +397,25 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
 		log.CtxDebugw(reqCtx.Context(), reqCtx.String())
 	}()
 
-	queryParams := r.URL.Query()
-	gnfdUserParam := queryParams.Get(GnfdUserAddressHeader)
-	gnfdOffChainAuthAppDomainParam := queryParams.Get(GnfdOffChainAuthAppDomainHeader)
-	gnfdAuthorizationParam := queryParams.Get(GnfdAuthorizationHeader)
-
-	// if all required off-chain auth headers are passed in as query params, we fill corresponding headers
-	if gnfdUserParam != "" && gnfdOffChainAuthAppDomainParam != "" && gnfdAuthorizationParam != "" {
-		r.Header.Set(GnfdUserAddressHeader, gnfdUserParam)
-		r.Header.Set(GnfdOffChainAuthAppDomainHeader, gnfdOffChainAuthAppDomainParam)
-		r.Header.Set(GnfdAuthorizationHeader, gnfdAuthorizationParam)
-	}
-
 	reqCtx, reqCtxErr = NewRequestContext(r, g)
+	if reqCtxErr != nil {
+		queryParams := r.URL.Query()
+		gnfdUserParam := queryParams.Get(GnfdUserAddressHeader)
+		gnfdOffChainAuthAppDomainParam := queryParams.Get(GnfdOffChainAuthAppDomainHeader)
+		gnfdOffChainAuthAppExpiryTimestampParam := queryParams.Get(commonhttp.HTTPHeaderExpiryTimestamp)
```

Comment:
I think the auth logic of `getObjectHandler`  is extremely complicated now. I will suggest following more straightforwad workflow:
1. No need for any further check if the object is public. 
2. if it is private then check if there is an authorization header, if yes go to personal sign and off chain auth workflow. if no, go to the query param auth workflow.
3. After that we get the `request sender`, do the `VerifyPermission`

Reply:
Done